### ISSUE TITLE
More than 32 Components

### ISSUE THREAD

**Issue by Norbert515** [denniskaselow/dartemis#53]
```
The following RangeError was thrown building FutureBuilder<void>(state: _FutureBuilderState<void>#e4ea7):
RangeError (index): Invalid value: Not in inclusive range 0..31: 32

The relevant error-causing widget was: [...]
When the exception was thrown, this was the stack: 
#0      List.[] (dart:core-patch/array.dart:110:52)
#1      Bag.[] (package:dartemis/src/core/utils/bag.dart:20:37)
#2      ComponentManager._registerSystem (package:dartemis/src/core/component_manager.dart:17:47)
#3      World.addSystem (package:dartemis/src/core/world.dart:124:22)
#4      ECSGame.onAttach (package:hacknslash/game/ECSGame.dart:119:11)
```

Hi, it seems that the current implementation doesn't allow for systems to reference more than 32 components. This seems to be the case because:
```
  ComponentManager._internal() : _componentInfoByType = Bag<_ComponentInfo>();
```

where the default `Bag` constructor has a capacity of 32. Am I understanding this correctly?

Would increasing the `Bag` capacity at that place resolve the issue?

Thanks!

**Comment by denniskaselow** [comment]
Interesting... https://github.com/denniskaselow/dartemis/blob/master/lib/src/core/component_manager.dart#L43 should increase the capacity if there are more than the default number of components. I guess you are adding systems but haven't created an entity with the components which the system is interested in. Which is totally fine and should work, so I'm going to fix that. Which version are you using?

**Comment by denniskaselow** [comment]
Fixed in version [0.9.1](https://pub.dev/packages/dartemis/versions/0.9.1). Leave a comment if you need a fix for 0.8.0.

**Comment by Norbert515** [comment]
Yeah, I'm registering all the systems before creating any entity. 0.9.1 fixes the issue for me, thank you very much!
